modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/FreeMarkerTool.java

```diff
@@ -1127,32 +1185,36 @@ public boolean isGeneratePermissions(
 
 		Map<String, Schema> propertySchemas = schema.getPropertySchemas();
 
+		JavaMethodSignature getPermissionsPageJavaMethodSignature =
+			getPermissionsPageJavaMethodSignature(
+				"get", javaMethodSignatures, schemaName);
+
+		JavaMethodSignature putPermissionsPageJavaMethodSignature =
+			getPermissionsPageJavaMethodSignature(
+				"put", javaMethodSignatures, schemaName);
+
+		String parentSchemaName = GetterUtil.getString(
```

Comment:
I think this is considering any other parent apart from AssetLibrary and Site, not sure if we need to support it